khelo/lib/ui/flow/tournament/detail/tournament_detail_view_model.dart

```diff
@@ -115,28 +116,95 @@ class TournamentDetailStateViewNotifier
     }
   }
 
-  void onStatFilter(KeyStatTag? tag) {
+  void onStatFilter(KeyStatFilterTag? tag) {
     if (state.tournament == null) return;
 
-    final filteredStats = state.tournament!.keyStats
-        .where((element) => element.tag == (tag ?? state.statFilter))
-        .toList();
+    var filteredStats = state.tournament!.keyStats;
+
+    switch (tag) {
+      case KeyStatFilterTag.runs:
+        filteredStats = filteredStats
+            .where((element) => (element.stats.battingStat?.runScored ?? 0) > 0)
```

Comment:
I see the pattern for where condition and sorting according to Tag, is there any way we can improve this to reduce code repetition?